### DISCUSSION TITLE
mpd error when upgarde and restart

### DISCUSSION THREAD

**Discussion by wan2355** [Homebrew/discussions#1656] [Everyday usage]
Yesterday, I have upgraded homebrew of my iMac, OSX 10.14.6, and following alert came up.

My system or "mpd" formula has some problem ?

What does it mean and what should I do to fix these issues ?

```
Warning: Some installed formulae are not readable:
  mpd: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:4: syntax error, unexpected <<, expecting end
<<<<<<< Updated upstream
^~
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:15: syntax error, unexpected ===, expecting end
=======
^~~
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:19: syntax error, unexpected <<, expecting end
<<<<<<< Updated upstream
^~
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:28: syntax error, unexpected ===, expecting end
=======
^~~
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:37: syntax error, unexpected >>, expecting end
>>>>>>> Stashed changes
^~
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:38: syntax error, unexpected >>, expecting end
>>>>>>> Stashed changes
^~
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:171: syntax error, unexpected end, expecting end-of-input
```

Moreover, I restart mpd service, same error comes up.
 

```
$ brew services restart mpd

Error: mpd: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:4: syntax error, unexpected <<, expecting end
<<<<<<< Updated upstream
^~
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:15: syntax error, unexpected ===, expecting end
=======
^~~
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:19: syntax error, unexpected <<, expecting end
<<<<<<< Updated upstream
^~
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:28: syntax error, unexpected ===, expecting end
=======
^~~
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:37: syntax error, unexpected >>, expecting end
>>>>>>> Stashed changes
^~
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:38: syntax error, unexpected >>, expecting end
>>>>>>> Stashed changes
^~
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/mpd.rb:171: syntax error, unexpected end, expecting end-of-input

```

**Comment by SMillerDev** [comment]
You made changes in the mpd formula and that caused issue. `brew update-reset` will reset them for you.

**Comment by wan2355** [comment]
Thank you.

I have forgotten modifying the formula....
I use original formula, errors disappear.

One more issue.

my mpd  require icu4c.
But someone says on the net that  
>OSX has same library "libicucore.dylib".
>So we do not need "icu4c" anymore.

If so, I like to know how to install mpd without icu4c.

**Reply by SMillerDev** [reply]
If brew made the formula depend on `icu4c` it is most likely because it didn't work with the macOS version. I would recommend against changing the dependencies.

If you do want to use `pmpd` without `icu4c` you can use [`brew extract`](https://docs.brew.sh/Manpage#extract---version---force-formula-tap) to make a copy of the formula in your own [tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap) so you can edit it however you want.

**Reply by wan2355** [reply]
I will try later.

>If you do want to use pmpd without icu4c you can use brew extract to make a copy of the formula in your own tap so you can edit it however you want.
